Modules/posixmodule.c

```diff
@@ -394,6 +395,87 @@ static int win32_can_symlink = 0;
 #define MODNAME "posix"
 #endif
 
+
+#ifdef HAVE_FORK
+static void
+run_at_forkers(PyObject *lst)
+{
+    if (lst != NULL) {
+        Py_ssize_t i;
+        assert(PyList_CheckExact(lst));
+        for (i = 0; i < PyList_GET_SIZE(lst); i++) {
+            PyObject *func, *res;
+            func = PyList_GET_ITEM(lst, i);
+            res = PyObject_CallObject(func, NULL);
+            if (res == NULL)
+                PyErr_WriteUnraisable(func);
+            else
+                Py_DECREF(res);
+        }
+    }
+}
+
+void
+PyOS_BeforeFork(void)
+{
+    /* NOTE callbacks are prepended in register_at_fork(), which is
```

Comment:
That's true. I didn't want to bother about this edge case, but a simple solution would be to copy the list before processing it.